src/v/transform/transform_processor.cc

```diff
@@ -231,6 +231,19 @@ processor::load_latest_committed() {
                               o.has_value() ? kafka::prev_offset(o.value())
                                             : latest);
                         });
+                  },
+                  [this](model::transform_from_start off) {
+                      vlog(_logger.debug, "starting at offset: {}", off);
+                      auto actual_offset = _source->start_offset() + off.delta;
+                      // We want to *start at* this offset, so record that we're
+                      // going to commit progress at the offset before, so we
+                      // start inclusive of this offset.
+                      return ssx::now(kafka::prev_offset(actual_offset));
+                  },
+                  [this, &latest](model::transform_from_end off) {
+                      vlog(_logger.debug, "starting at offset: {}", off);
+                      auto actual_offset = latest - off.delta;
```

Comment:
underflow

Reply:
sigh. goood catch. thought i made a note somewhere to address that 😕